trace/trace.go

```diff
@@ -582,6 +582,26 @@ type Link struct {
 	DroppedAttributeCount int
 }
 
+// MarshalJSON implements a custom marshal function to encode a Link.
+// This is a workaround to correctly marshal Link. Without this custom
+// MarshalJSON method, fields other than SpanContext will not be
+// marshalled.
+func (l Link) MarshalJSON() ([]byte, error) {
+	lJSON, err := json.Marshal(struct {
+		SpanContext           SpanContext
+		Attributes            []attribute.KeyValue
+		DroppedAttributeCount int
+	}{
+		SpanContext:           l.SpanContext,
+		Attributes:            l.Attributes,
+		DroppedAttributeCount: l.DroppedAttributeCount,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return lJSON, nil
```

Comment:
I wonder if this can just `return json.Marshal(...` directly.  Does `json.Marshal` return a partially populated byte slice in the error case?  If not, this check should be unnecessary.

Reply:
Yes. I have verified that, as #1820 described, if you call `json.Marshal(trace.Link{...})` directly, it will only return the `SpanContext` part.

Reply:
@IrisTuntun,  @Aneurysm9  actually means  it's unnecessary to do this `if err != nil`,  doing just like this: `func (l Link) MarshalJSON() ([]byte, error) {
	return json.Marshal(struct {
		SpanContext           SpanContext
		Attributes            []attribute.KeyValue
		DroppedAttributeCount int
	}{
		SpanContext:           l.SpanContext,
		Attributes:            l.Attributes,
		DroppedAttributeCount: l.DroppedAttributeCount,
	})
}`